File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/inheritance/callingSuperClassImpl/Derived.java
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.callingSuperClassImpl;

/**
 * Demonstrates the calling of super class implementation
 *
 * @author Ranjith Krishnamurthy
 */
public class Derived extends Base {
    @Override
    public void print() {
        super.print();
        System.out.println("Derived");
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/inheritance/callingSuperClassImpl/Derived.kt
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.callingSuperClassImpl

/**
 * Demonstrates the calling of super class implementation
 *
 * @author Ranjith Krishnamurthy
 */
class Derived : Base() {
    override fun print() {
        super.print()   // Super class implementation is called
        println("Derived")
    }
}